controllers/provisioning_controller.go

```diff
@@ -36,9 +38,11 @@ const (
 type ProvisioningReconciler struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	Client client.Client
-	Scheme *runtime.Scheme
-	Log    logr.Logger
+	Client        client.Client
+	Scheme        *runtime.Scheme
+	Log           logr.Logger
+	osClient      osclientset.Interface
```

Comment:
I don't see anything populating `osClient`. Where does that happen?

Reply:
This one needs to go when we initialize the reconciler. Not sure whether the reconcile code is in a sturdy state yet to add stuff there...

Reply:
Added this logic in the latest commit.